### PR TITLE
tflite: python: tflitePythonVivante: Fix build process and remove pre-built libraries

### PR DESCRIPTION
**Pull request by brunoaamello**
Hello maintainers,

I am creating this pull request to hopefully solve the build issues present in the tflitePythonVivante sample.

This PR also:
- Removed the pre-built libraries for TensorFlow Lite
- Added missing library build steps to release container

This is initially set as a draft because I want to do some more testing to make sure everything works as expected.

### CODE REVIEW

**Comment by leograba** [comment]
I see a change in style, for example on Dockerfiles instead of keeping one package per line on `RUN apt install` statements, moving to several packages per line.

Not nitpicking on them as there are no agreed rules on style. Just pointing it our so it does not go unnoticed.

**Comment by leograba** [comment]
@lucasbernardestoradex since you have recently worked on this topic, I'd appreciate you review the changes as well and let us know your thoughts.

**Comment by charles2910** [comment]
> I see a change in style, for example on Dockerfiles instead of keeping one package per line on `RUN apt install` statements, moving to several packages per line.
> 
> Not nitpicking on them as there are no agreed rules on style. Just pointing it our so it does not go unnoticed.

I would prefer to keep them listed one per line for 2 main reasons (and tabs set to 4 spaces because it seems to what the  IDE uses), it's easier to parse the package list that way and it's also easier to see diffs. If you add or remove a package, you get exactly that in the diff. About the other changes in style, I don't have a preferred style, so it's good as is.

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    python3 \
    python3-dev \
    python3-numpy \
    python3-pybind11 \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    python3-pip \
    python3-setuptools \
    python3-wheel \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    cmake \
    build-essential \
    gcc \
    g++ \
    git \
    wget \
    unzip \
    patchelf \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    autoconf \
    automake \
    libtool \
    curl \
    gfortran && \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]: The shared lib in Debian is always a dependency of the `-dev` package, so we can drop the `zlib1g` package.

```suggestion
    zlib1g-dev \
    libssl-dev \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    imx-gpu-viv-wayland-dev \
    openssl \
    libffi-dev \
    libjpeg-dev && \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]:
```suggestion
    wget \
    unzip \
    imx-gpu-viv-wayland-dev \
```

File: tflite/python/tflitePythonVivante/Dockerfile
Comment (charles2910) [inline tflite/python/tflitePythonVivante/Dockerfile]: could you please add the final \n in line 102?

**Review by charles2910 — COMMENTED** [review]
Nice, thanks for the work. I'm building the container image just to be sure, but I suspect it's going to take a while to finish

**Comment by andreriesco** [comment]
Hello @brunoaamello, since this template has been added a while ago, can you also [update the IDE part of the template](https://developer.toradex.com/torizon/application-development/ide-extension/updating-projects-by-template)?

**Comment by brunoaamello** [comment]
Thanks for the comments @leograba, @charles2910 and @andreriesco.

I will apply the suggested changes and update the IDE template as requested.

**Review by lucas-p-bernardes — COMMENTED** [review]
I did a quick review and the changes seem reasonable to me.
I was having a lot of warning problems with `tim-vx.sh`, which I believe were fixed with these changes.

**Comment by brunoaamello** [comment]
Hello maintainers,

Thanks for your comments and sorry for the delay.

I have updated the Pull Request according to them and tested both the debug and release containers.
Everything works as expected with NPU acceleration.

I will remove the "Draft" from the title as I believe it is ready to be reviewed and merged.

File: tflite/python/tflitePythonVivante/.conf/createDockerComposeProduction.ps1
Comment (andreriesco) [inline tflite/python/tflitePythonVivante/.conf/createDockerComposeProduction.ps1:155]: I added a little fix on this file. Can you add that fix, please? You don't need to run the `try-update-template` task again, as it is a small change you can add it manually in this file.
[https://github.com/toradex/vscode-torizon-templates/commit/9195cdb9e84ecf7ec1cc3600f105017a573e08ae](https://github.com/toradex/vscode-torizon-templates/commit/9195cdb9e84ecf7ec1cc3600f105017a573e08ae)

File: tflite/python/tflitePythonVivante/.vscode/settings.json
Comment (leograba) [inline tflite/python/tflitePythonVivante/.vscode/settings.json]: I'm not sure whether it's good or bad to leave pre-filled fields in this config file. Leaving a comment for it to be evaluated, just in case.

Comment (andreriesco) [reply tflite/python/tflitePythonVivante/.vscode/settings.json]: Actually they should be removed. You can do it by removing the "Default Device" from the IDE interface. It will automatically update this parameters to empty values: [https://developer.toradex.com/torizon/application-development/use-cases/samples/#how-to-contribute](https://developer.toradex.com/torizon/application-development/use-cases/samples/#how-to-contribute)

File: tflite/python/tflitePythonVivante/.conf/createDockerComposeProduction.ps1
Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/createDockerComposeProduction.ps1:155]: Surely, this is done.

If I misunderstood exactly what needed to be done, please let me know.

File: tflite/python/tflitePythonVivante/.vscode/settings.json
Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.vscode/settings.json]: I completely agree.
Sorry for that, I forgot to remove those.
Now they have been removed.

**Review by charles2910 — APPROVED** [review]
Chef's kiss

**Review by leograba — APPROVED** [review]
Molto bene!

File: tflite/python/tflitePythonVivante/.conf/deps.json
Comment (andreriesco) [inline tflite/python/tflitePythonVivante/.conf/deps.json:9]: I think we need to keep `python3-numpy` here, since the application needs it.

Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: @andreriesco, I think numpy is installed with Tensoflow Lite here: https://github.com/toradex/torizon-samples/pull/18/files/814c551748f5b5ce5760fdd97b3a6063ef3da238#diff-7ce7be0f090719e6ecb5560365d3c08aea233046719f83a4e074b9267e2c2d1fR96

It is one of the dependencies of the TensorFlow lite wheel: https://github.com/nxp-imx/tensorflow-imx/blob/lf-5.15.71_2.2.0/tensorflow/lite/python/BUILD#L38

Comment (andreriesco) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: Actually this packages present in the `.conf/deps.json` are the ones that are installed on the host machine. So, as the application needs numpy to be executed on Local Debug, it would need numpy to be installed on the host

Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: @andreriesco, I believe that what is installed on the host when you try to run the project locally is the contents of the requirements files.
From my testing, it even does that to the local python system packages with `/bin/python3 -m pip install --break-system-packages -r requirements-debug.txt`, which is quite dangerous.

Given that it is this way, there are two paths forward:
- Add the required dependencies to the `requirements*.txt` files and test if this does not break the container on the module.
- Only support the containers to run on the module

My preference would be to only support the containers to run on the module, as such installation of packages to the local python system packages is not acceptable in my opinion.

Comment (andreriesco) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: The numpy in specific is actually being installed through an `apt` package [on the container](https://github.com/toradex/torizon-samples/pull/18/commits/814c551748f5b5ce5760fdd97b3a6063ef3da238#diff-be950562ae9442005cab237cc42705ee3bd9c2c999116268c9b2c17aeea14351R41), so that's why it is on `deps.json`. 
About the installation of the pip packages with this command `/bin/python3 -m pip install --break-system-packages -r requirements-debug.txt`, probably we should review it and improve it. But not just for this sample, for the whole python3 IDE template. Can you open a thread about this on the Slack channel of the IDE so we can discuss better this problem and possible solutions, please?

Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: > About the installation of the pip packages with this command /bin/python3 -m pip install --break-system-packages -r requirements-debug.txt, probably we should review it and improve it. But not just for this sample, for the whole python3 IDE template. Can you open a thread about this on the Slack channel of the IDE so we can discuss better this problem and possible solutions, please?

Surely, I will open a discussion there.


> The numpy in specific is actually being installed through an apt package [on the container](https://github.com/toradex/torizon-samples/pull/18/commits/814c551748f5b5ce5760fdd97b3a6063ef3da238#diff-be950562ae9442005cab237cc42705ee3bd9c2c999116268c9b2c17aeea14351R41), so that's why it is on deps.json.

The numpy package is being installed on the build stage of the debug container. It is a build dependency for the tflite python package. This dependency gets then added to the wheel which is resolved by pip on the application container.

Comment (andreriesco) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: Sounds ok for now to keep the usage just inside the containers. Then, after we change the default behavior of the Python template I believe we can update the sample with this change and then allow the local debugging also

Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: Hello @andreriesco,

Sorry for bringing back this topic, I was working on enabling the local debugging and did not see your comment.

I added support for it with the following commit: https://github.com/toradex/torizon-samples/pull/18/commits/e05f7a325bb183dbcc3ce48abd02a734352a7a63

It still is not ideal if the user tries to run it locally because of the issues we discussed, but it should work without problems within a venv.

Please let me know if you would like me to revert the commit or if we can keep it as is.

Comment (brunoaamello) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: Note: the addition of [requirements-debug-local.txt](https://github.com/toradex/torizon-samples/commit/e05f7a325bb183dbcc3ce48abd02a734352a7a63#diff-094433cd1023e5e21c0e5c930da22e613a18f6e7e8d2fd1d243924033f617be9) is intended to avoid any unnecessary installation of packages in the container.

Comment (andreriesco) [reply tflite/python/tflitePythonVivante/.conf/deps.json:9]: Nice, I think we can keep it as it is. Probably I will something similar to this on the python3 IDE template, so I think it makes sense to keep it.

**Comment by andreriesco** [comment]
LGTM

**Pull request merged**
